packages/@lwc/engine/dom/src/env/shadow-dom.ts

```diff
@@ -0,0 +1,9 @@
+/*
+ * Copyright (c) 2020, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: MIT
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/MIT
+ */
+import { hasOwnProperty } from '@lwc/shared';
+
+export const useSyntheticShadow = hasOwnProperty.call(Element.prototype, '$shadowToken$');
```

Comment:
If this is the only property, you could move it to `/env/dom.ts`, synthetic-shadow is a character of the dom.